reckon-core/src/main/java/org/ajoberstar/reckon/core/Versions.java

```diff
@@ -60,4 +60,23 @@ public static Version incrementNormal(Version version, Scope scope) {
         throw new AssertionError("Invalid scope: " + scope);
     }
   }
+
+  public static Optional<Scope> inferScope(Version before, Version after) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Document this public method by adding an explicit description. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3AUndocumentedApi)